deployment/helm/templates/ingress.yaml

```diff
@@ -0,0 +1,49 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  annotations:
+    # add an annotation indicating the issuer to use.
+    cert-manager.io/acme-challenge-type: http01
+    cert-manager.io/cluster-issuer: letsencrypt-prod
+  name: {{ .Values.settings.name }}-{{ .Values.settings.env }}-ingress
+  labels:
+    app.kubernetes.io/name: {{ .Values.settings.name }}-ingress
+    app.kubernetes.io/part-of: {{ .Values.settings.name }}
+    helm.sh/chart: {{ include "chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  tls:
+    - hosts:
+        - {{ $.Values.settings.domain }}
+      secretName: {{ template "fullname" . }}
+  rules:
+    {{- if and (eq $.Values.settings.env "prod") $.Values.settings.subdomain }}
+
+    - host: {{ $.Values.settings.subdomain }}.{{ $.Values.settings.domain }}
+      http:
+        paths:
+            - backend:
+                serviceName: {{ $.Values.settings.name }}
+                servicePort: 80
+    {{- else if eq $.Values.settings.env "prod" }}
+    - host {{ $.Values.settings.domain }}
+      http:
+        paths:
+            - backend:
+                serviceName: {{ $.Values.settings.name }}
+                servicePort: 80
+    - host www.{{ $.Values.settings.domain }}
+      http:
+        paths:
+            - backend:
+                serviceName: {{ $.Values.settings.name }}
+                servicePort: 80
+    {{- else }}
+    - host: {{ $.Values.settings.subdomain }}.{{ $.Values.settings.env }}.{{ $.Values.settings.domain }}
+      http:
+        paths:
+            - backend:
+                serviceName: {{ $.Values.settings.name }}
+                servicePort: 80
+    {{- end }}
```

Comment:
I would strongly advise not to have multiple domains pointing to the same instance, because you then get different URLs pointing to the same resource in the API, which in turns can break relation-filtering in other systems that save references via URL (as they should!)